x/leveragelp/keeper/position_open.go

```diff
@@ -113,9 +94,11 @@ func (k Keeper) ProcessOpenLong(ctx sdk.Context, position *types.Position, lever
 
 	// borrow leveragedAmount - collateralAmount
 	borrowCoin := sdk.NewCoin(msg.CollateralAsset, leveragedAmount.Sub(msg.CollateralAmount))
-	err = k.stableKeeper.Borrow(ctx, position.GetPositionAddress(), borrowCoin)
-	if err != nil {
-		return nil, err
+	if borrowCoin.Amount.GT(sdk.ZeroInt()) {
+		err = k.stableKeeper.Borrow(ctx, position.GetPositionAddress(), borrowCoin)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think we are missing repayment in this flow, Case:
- User opens a position with 2x lev
- Now wants to repay by adding collateral, so will open position with 1x lev
- That should repay the liability(we are not borrowing here but need to repay)

Reply:
wait, that might not be it. what do you think should happen in this case ?
another question is what's the flow for adding collateral now ?

Reply:
got it, but we need repay to improve health, so you need to repay amount passed by user when leverage is 1

Reply:
refer to add collateral flow

Reply:
@cosmic-vagabond 
1. got it
2. makes sense
     a. I was thinking to reduce debt, as it is more impactful on position health
     b. For examples: a position with LevAmount = 200$ and Debt = 100$
     c  User adds, 50$, If we consolidate, health moves from 2 -> 2.5
     d. But if we repay, health moves from 2 -> 4
But this is fine too because health has increased in both cases.

Reply:
@amityadav0 the only issue I might see is if the interest payment relies on the position settings, do we need to settle the interest payment when we call the open consolidate flow, I am not sure if that’s already the case.

Reply:
as of now, we don't settle interest payment on this flow. okay, we can keep it as is.